generators/kubernetes-knative/files.js

```diff
@@ -0,0 +1,168 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module.exports = {
+    writeFiles
+};
+
+function writeFiles() {
+    return {
+        writeGeneratorFiles() {
+            const k8s = this.fetchFromInstalledJHipster('kubernetes/templates');
+            const helm = this.fetchFromInstalledJHipster('kubernetes-helm/templates');
+            if (this.kubernetesNamespace !== 'default') {
+                this.template(`${k8s}/namespace.yml.ejs`, 'namespace.yaml');
+            }
+            this.template('README-KUBERNETES-KNATIVE.md.ejs', 'README.md');
+            if (this.generatorType === 'k8s') {
+                for (let i = 0; i < this.appConfigs.length; i++) {
+                    const appName = this.appConfigs[i].baseName.toLowerCase();
+                    this.app = this.appConfigs[i];
+                    this.template('service.yml.ejs', `${appName}/${appName}-service.yaml`);
+                    // If we choose microservice with no DB, it is trying to move _no.yml as prodDatabaseType is getting tagged as 'string' type
+                    if (this.app.prodDatabaseType !== 'no') {
+                        this.template(
+                            `${k8s}/db/${this.app.prodDatabaseType}.yml.ejs`,
+                            `${appName}/${appName}-${this.app.prodDatabaseType}.yaml`
+                        );
+                    }
+                    if (this.app.searchEngine === 'elasticsearch') {
+                        this.template(`${k8s}/db/elasticsearch.yml.ejs`, `${appName}/${appName}-elasticsearch.yaml`);
+                    }
+                    if (this.app.applicationType === 'gateway' || this.app.applicationType === 'monolith') {
+                        this.template('istio/gateway.yml.ejs', `${appName}/${appName}-gateway.yaml`);
+                    }
+                    if (!this.app.serviceDiscoveryType && this.app.authenticationType === 'jwt') {
+                        this.template(`${k8s}/secret/jwt-secret.yml.ejs`, `${appName}/jwt-secret.yaml`);
+                    }
+                    if (this.monitoring === 'prometheus') {
+                        this.template(`${k8s}/monitoring/jhipster-prometheus-sm.yml.ejs`, `${appName}/${appName}-prometheus-sm.yaml`);
+                    }
+                    this.template('istio/destination-rule.yml.ejs', `${appName}/${appName}-destination-rule.yaml`);
+                    this.template('istio/virtual-service.yml.ejs', `${appName}/${appName}-virtual-service.yaml`);
+                }
+
+                if (this.useKafka) {
+                    this.template(`${k8s}/messagebroker/kafka.yml.ejs`, 'messagebroker/kafka.yaml');
```

Comment:
nit: for the `windows` folks and to maintain consistency can we use `.yml` instead of `.yaml`